File: Easy/28FirstNonRepeatingCharacter.py
def firstNonRepeatingCharacter(string):
    # Write your code here.
    characterFrequincies = {}
    for i in string:
        characterFrequincies[i] = characterFrequincies.get(i, 0)+1

    for idx in range(len(string)):
        if characterFrequincies[string[idx]] == 1:
            return idx
    return -1


print(firstNonRepeatingCharacter("abcdcaf"))

File: Easy/28FirstNonRepeatingCharacter.js
function firstNonRepeatingCharacter(string) {
  // Write your code here.
  let characterFrequencies = {};
  for (let i of string) {
    if (!characterFrequencies.hasOwnProperty(i)) characterFrequencies[i] = 0;
    characterFrequencies[i] += 1;
  }
  for (let i = 0; i < string.length; i++) {
    if (characterFrequencies[string[i]] == 1) return i;
  }
  return -1;
}

console.log("====================================");
console.log(firstNonRepeatingCharacter("abcdcaf"));
console.log("====================================");
